contentcuration/contentcuration/utils/publish.py

```diff
@@ -538,6 +543,20 @@ def process_image_strings(content, zf):
             # Add any image files that haven't been written to the zipfile
             filename = img_match.group(1).split('/')[-1]
             checksum, ext = os.path.splitext(filename)
+
+            if not ext:
+                logging.warning("While publishing channel `{}` a filename with no extension was encountered: `{}`".format(channel_id, filename))
+            try:
+                # make sure the checksum is actually a hex string
+                int(checksum, 16)
+            except Exception:
+                logging.warning("while publishing channel `{}` a filename with an improper checksum was encountered: `{}`".format(channel_id, filename))
+
+                # if we're in a testing or development environment, raise the error
+                if os.environ.get('BRANCH_ENVIRONMENT', '') != "master":
```

Comment:
@rtibbles as it turns out, this is something Kevin had already added [in this line](https://github.com/learningequality/studio/pull/2922/files#diff-80dad321db1514a4994051563fa1efbd77dafdb6b679fb5e00c93e8eb37aeec9R458).  I verified that locally when running tests, `os.environ.get('BRANCH_ENVIRONMENT', '')` is `None`.